### PR TITLE
Vendor k/* to v0.21.1 and c-r dependencies to v0.9.0

### PR DESCRIPTION
**Pull request by timebertt**
**How to categorize this PR?**

/area open-source
/kind task

**What this PR does / why we need it**:

Vendors `k8s.io/*@v0.21.1` and `sigs.k8s.io/controller-runtime@v0.9.0` and adapts to the breaking / most notable changes (see https://github.com/gardener/gardener/issues/4174#issuecomment-862098004).

**Which issue(s) this PR fixes**:
Main part of #4174, a few optional cleanups will follow in dedicated PRs to keep this one limited to the minimum scope.

**Special notes for your reviewer**:

To ease the review, changes are split into dedicated commits, each targeting "one step" / "one adaption" to an upstream change.
Commits marked with `[automated]` are results of `make *` with exception of minimal manual changes described in the commit message.
/squash

**Release note**:

```other dependencies
The following golang dependencies have been upgraded, please consult the upstream release notes and [this note](https://github.com/gardener/gardener/issues/4174#issuecomment-862098004) for guidance on upgrading your golang dependencies when vendoring this gardener version:
- `k8s.io/*` to `v0.21.1`
- `sigs.k8s.io/*` to `v0.9.0`
```

### CODE REVIEW

File: pkg/envtest/aggregator.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+// ConfigureAPIServerArgs generates the needed certs, writes them to the given directory and configures args
+// to point to the generated certs.
+func (a AggregatorConfig) ConfigureAPIServerArgs(certDir string, args *envtest.Arguments) error {
+	a.certDir = certDir
+
+	if err := a.generateCerts(); err != nil {
+		return err
+	}
+
+	args.Set("requestheader-extra-headers-prefix", "X-Remote-Extra-").
```
Comment (rfranzke) [inline pkg/envtest/aggregator.go]:
```suggestion
	args.
		Set("requestheader-extra-headers-prefix", "X-Remote-Extra-").
```

style nit :)

**Review by rfranzke — DISMISSED** [review]
well structured and well done!
/lgtm

File: pkg/envtest/aggregator.go (hunk 1 above)
Comment (timebertt) [reply pkg/envtest/aggregator.go]: Done :)

**Review by rfranzke — APPROVED** [review]

**Comment by timebertt** [comment]
Thanks for the fast review, @rfranzke! ❤️

File: pkg/envtest/aggregator.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package envtest
+
+import (
+	"fmt"
+	"io/ioutil"
```
Comment (ialidzhikov) [inline pkg/envtest/aggregator.go:19]: Nit: with https://github.com/gardener/gardener/issues/4020 we adapted all `io/utils` usage. Ref https://github.com/gardener/gardener/pull/4221.

File: Makefile [hunk 3]
```diff
@@ -164,7 +165,6 @@ install-requirements:
 revendor:
 	@GO111MODULE=on go mod vendor
 	@GO111MODULE=on go mod tidy
-	@curl -sSLo hack/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/$(CR_VERSION)/hack/setup-envtest.sh
```
Comment (ialidzhikov) [inline Makefile:167]: Nit: `CR_VERSION` var can be also removed

**Review by ialidzhikov — COMMENTED** [review]
Thanks for the PR @timebertt !

File: pkg/envtest/aggregator.go (hunk 2 above)
Comment (timebertt) [reply pkg/envtest/aggregator.go:19]: Ah, thanks for noticing!

File: Makefile (hunk 3 above)
Comment (timebertt) [reply Makefile:167]: Nice catch!

**Pull request merged**